tests/core/test_metric_result_integration.py

```diff
@@ -132,13 +132,28 @@ def test_result_metric_integration():
 
         assert epoch_log == {"b": cumulative_sum, "a_epoch": cumulative_sum}
 
+    result.minimize = torch.tensor(1.0)
+    result.extra = {}
     assert str(result) == (
-        "ResultCollection(True, cpu, {"
+        "ResultCollection("
+        "minimize=1.0, "
```

Comment:
Personal taste, but I prefered the previous one :)

Reply:
We still have the previous one as `__repr__`.

I found this splitting these made debugging easier. You don't want to show all the details in `__str__`